src/mpfao/3D/tdympfao3D_ts.c

```diff
@@ -75,8 +77,15 @@ PetscErrorCode TDyMPFAOIFunction_Vertices_3DMesh(Vec Ul, Vec R, void *ctx) {
       cell_id_up = faces->cell_ids[fOffsetCell + 0];
       cell_id_dn = faces->cell_ids[fOffsetCell + 1];
 
+      den = 0.0;
+      if (cell_id_up>=0) den += tdy->rho[cell_id_up];
```

Comment:
This style is a little loose for C. Would you mind doing something like the following?

```
if (cell_id_up>=0) {
  den += tdy->rho[cell_id_up];
} else {
  den += tdy->rho_BND[-cell_id_up-1];
}